x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/persistence/TrainedModelProvider.java

```diff
@@ -216,6 +219,28 @@ public void storeTrainedModelConfig(TrainedModelConfig trainedModelConfig, Actio
         );
     }
 
+    /**
+     * Updates a model's {@link TrainedModelConfig#MODEL_SIZE_BYTES} to a new value
+     * @param modelId the model id for the document to update
+     * @param modelSize the new value for the model size
+     * @param listener the listener
+     */
+    public void updateModelSize(String modelId, long modelSize, ActionListener<UpdateResponse> listener) {
+        SearchRequest searchRequest = buildTrainedModelConfigSearch(modelId);
```

Comment:
Shouldn't you set `"seq_no_primary_term": true,`? You then need to set both the term and seqno on the update request I think.